aioredis/connection.py

```diff
@@ -503,12 +537,6 @@ async def read_response(self) -> EncodableT:
         # happened
         if isinstance(response, ConnectionError):
             raise response
-        elif (
```

Comment:
@Dreamsorcerer I'm guessing this is the problem. Add this elif branch back in and the error should now be raised correctly.

Pipelines return a series of responses, not a single response, so we have to inspect the series for an error.

Reply:
Ah, I didn't look closely enough at the annotations. It's `self._next_response` which defines the type as `bool`, and the hiredis annotation is `Any`, so fails to highlight the correct issue to us.

Reply:
OK, so it checks if it is a list starting with a `ConnectionError`, but what if it's a list with some other value? The return type is `EncodableT` which does not include a list of anything. Should `EncodableT` include a list, or should the return type be something like `Union[EncodableT, List[EncodableT]]`?

Reply:
At first glance, I'd say do the Union, but iirc that didn't play nicely with the assumptions in the standard client that it's never a list. Union doesn't flag to mypy the case in which a List may occur.